### PR TITLE
[WFLY-11674] JTA documentation to be enriched with a section about synchronization

### PR DESCRIPTION
**Pull request by ochaloup**
https://issues.jboss.org/browse/WFLY-11674

addition to the JTA documentation article about how to register synchronization

### CODE REVIEW

File: docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc [hunk 1]
```diff
@@ -236,6 +236,102 @@ NOTE: if a call is made from the container-managed method,
...
+  public void afterCompletion(int status) {
+    System.out.println("Transaction finished with status " + status):
+  }
+}
+----
+
+For registration of the synchronization callback the user can inject the
+https://docs.oracle.com/javaee/7/api/javax/transaction/TransactionSynchronizationRegistry.html[javax.transaction.TransactionSynchronizationRegistry].
+(the mandated JNDI location for the object is at `java:comp/TransactionSynchronizationRegistry`) and then to register
+the sychronization instance. The instance is bound to the currently active transaction.
```
Comment (yersan) [inline docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc]: s/sychronization/synchronization

File: docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc [hunk 2]
```diff
@@ -236,6 +236,102 @@ NOTE: if a call is made from the container-managed method,
...
+@Resource
+TransactionSynchronizationRegistry transactionSynchronizationRegistry;
+
+public void method() {
+  transactionSynchronizationRegistry
+    .registerInterposedSynchronization(new MySynchronization());
+}
+----
+
+The transaction synchronization registry adds another useful methods which are `putResource(Object key, Object value)`
```
Comment (yersan) [inline docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc]: s/adds another useful methods/adds other useful methods

File: docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc [hunk 3]
```diff
@@ -236,6 +236,102 @@ NOTE: if a call is made from the container-managed method,
...
+TransactionManager tm;
+
+public void method() {
+tm.getTransaction().registerSynchronization(new MySynchronization());
+----
+
+When the user runs the Stateful Session Bean he can implement
+interface https://javaee.github.io/javaee-spec/javadocs/javax/ejb/SessionSynchronization.html[javax.ejb.SessionSynchronization]
+(or to use annotations) for definition of the synchronization callbacks onto the bean.
+The session synchronization defines three methods. The one additional method is `afterBegin`.
```
Comment (yersan) [inline docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc]: s/The one/One

File: docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc [hunk 4]
```diff
@@ -236,6 +236,102 @@ NOTE: if a call is made from the container-managed method,
...
+That one is not connected with the transaction synchronization thus we are leaving it aside.
+The  following example works with annotations but the bean may just implement the `SessionSynchronization`
+interface it would work the same way.
+
+[source,java,options="nowrap"]
+----
+// only SFSB can be useing the capability of SessionSynchronization
```
Comment (yersan) [inline docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc]: s/can be useing/can use

File: docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc [hunk 5]
```diff
@@ -236,6 +236,102 @@ NOTE: if a call is made from the container-managed method,
...
+
+public void method() {
+  transactionSynchronizationRegistry
+    .registerInterposedSynchronization(new MySynchronization());
+}
+----
+
+The transaction synchronization registry adds another useful methods which are `putResource(Object key, Object value)`
+and `getResource(Object key)`. Their purpose is saving data objects alongside the transaction context.
+When the transaction is active you can store and retrieve the saved data. When transaction is finished
```
Comment (yersan) [inline docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc]: s/When transaction/When the transaction

File: docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc [hunk 6]
```diff
@@ -236,6 +236,102 @@ NOTE: if a call is made from the container-managed method,
       call transaction managed bean annotated with
       `@TransactionAttribute(TransactionAttributeType.NOT_SUPPORTED)`
 
+==== Transaction synchronization
+
+JTA API gives a chance to react on event of a finishing transaction.
```
Comment (yersan) [inline docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc]: s/gives a chance to react on event of/gives a chance to react to the event of

File: docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc [hunk 7]
```diff
@@ -236,6 +236,102 @@ NOTE: if a call is made from the container-managed method,
...
+The definition says that transaction manager announces event of `beforeCompletion` and `afterCompletion`
```
Comment (yersan) [inline docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc]: s/announces event/announces the event

File: docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc [hunk 8]
```diff
@@ -236,6 +236,102 @@ NOTE: if a call is made from the container-managed method,
...
+  }
+  
+  @Override
+  public void afterCompletion(int status) {
+    System.out.println("Transaction finished with status " + status):
+  }
+}
+----
+
+For registration of the synchronization callback the user can inject the
```
Comment (yersan) [inline docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc]: s/the synchronization callback the user/the synchronization callback, the user

**Review by yersan — CHANGES_REQUESTED** [review]
I found some minor typo errors and added really minor details that make it sounds better to me

**Comment by ochaloup** [comment]
hi @yersan , thanks a lot for your review. I will try to do better next time :-)

but, for making my life easier :laughing:  (take it as a joke, I consider my errors as a silly typo issues that I should avoid next time) you can consider using suggestions (https://thenextweb.com/dd/2018/10/16/github-launches-suggested-changes-to-make-it-easier-to-collaborate-within-pull-requests/). I can apply your suggestiong directly to the PR with one click in the github UI.

**Comment by yersan** [comment]
Thanks @ochaloup I didn't know about the possibility of using suggestions, I'll use them next time

**Review by yersan — APPROVED** [review]

File: docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc [hunk 9]
```diff
@@ -236,6 +236,102 @@ NOTE: if a call is made from the container-managed method,
...
+TransactionManager tm;
+
+public void method() {
+tm.getTransaction().registerSynchronization(new MySynchronization());
+----
+
+When the user runs the Stateful Session Bean he can implement
+interface https://javaee.github.io/javaee-spec/javadocs/javax/ejb/SessionSynchronization.html[javax.ejb.SessionSynchronization]
+(or to use annotations) for the definition of the synchronization callbacks onto the bean.
+The session synchronization defines three methods. One additional method is `afterBegin`.
```
Comment (kabir) [inline docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc]: Looking at the interface mentioned, I would say  something like:
Of these three methods `afterBegin` is not connected to the transaction synchronization so we will not discuss it further.

Comment (ochaloup) [reply docs/src/main/asciidoc/_developer-guide/JTA_Reference.adoc]: @kabir perfect, thanks, changed.

**Pull request merged**